plugins/core/hu.bme.mit.gamma.ui/src/hu/bme/mit/gamma/ui/taskhandler/CodeGenerationHandler.java

```diff
@@ -41,27 +53,91 @@ public void execute(CodeGeneration codeGeneration, String packageName) {
 		// Setting target folder
 		setProjectLocation(codeGeneration); // Before the target folder
 		setTargetFolder(codeGeneration);
-		setCodeGeneration(codeGeneration, packageName);
 		//
-		List<ProgrammingLanguage> programmingLanguages = codeGeneration.getProgrammingLanguages();
-		checkArgument(programmingLanguages.size() == 1,
-				"A single programming language must be specified: " + programmingLanguages);
-		ProgrammingLanguage programmingLanguage = programmingLanguages.get(0);
+		checkArgument(codeGeneration.getProgrammingLanguages().size() == 1, 
+				"A single programming language must be specified: " + codeGeneration.getProgrammingLanguages());
+		checkArgument(codeGeneration.getProgrammingLanguages().get(0) == ProgrammingLanguage.JAVA ||
+				codeGeneration.getProgrammingLanguages().get(0) == ProgrammingLanguage.C,
+				"Currently only Java and C supported.");
+		setCodeGeneration(codeGeneration, packageName);
 		
-		switch (programmingLanguage) {
-			case JAVA:
-				generateJavaCode(codeGeneration);
-				break;
-			case C:
-				generateCCode(codeGeneration);
-				break;
-			default:
-				throw new IllegalArgumentException("Not known programming language: " + programmingLanguage);
+		switch(codeGeneration.getProgrammingLanguages().get(0)) {
```

Comment:
Everything is the same with the exception that `testGeneration.getProgrammingLanguages().get(0)` is not extracted into into a local variable. Could you revert these parts?